EIPS/eip-5299.md

```diff
@@ -0,0 +1,165 @@
+        // Set the position of our struct in contract storage
+        assembly {ps.slot := storagePosition}
+    }
+
+    // Returns the struct from a specified position in contract storage
+    // as is short for AppStorage
+    function appStorageV1() internal pure returns(AppStorageV1 storage as) {
+        bytes32 storagePosition = keccak256("app.storage.implContractA.V1");
+        assembly {as.slot := storagePosition}
+    }
+
+    // Returns the struct from a specified position in contract storage
+    // as is short for AppStorage
+    function appStorageV2() internal pure returns(AppStorageV2 storage as) {
+        bytes32 storagePosition = keccak256("app.storage.implContractA.V2");
+        assembly {as.slot := storagePosition}
+    }
+}
+
+// Our implementation contract uses the Parent Storage defined above along with the implementation contract specific storage
+contract ImplContractAV2 {
+    function setDataAV2(uint256 _dataA) external {
+        LibB.DiamondStorage storage ds = LibA.parentStorage();
+        LibB.AppStorageV2 storage as = LibA.appStorageV2();
+        require(ds.owner == msg.sender, "Must be owner.");
+        as.dataA = _dataA;
+  }
+
+    function getDataAV2() external view returns (uint256) {
+        LibB.AppStorageV2 storage as = LibA.appStorageV2();
+        return as.dataA;
+    }
+
+    function getDataAV1AndV2() external view returns (bytes32, uint256) {
+        LibB.AppStorageV1 storage as1 = LibA.appStorageV1();
+        LibB.AppStorageV2 storage as2 = LibA.appStorageV2();
+        return (as1.dataA, as2.dataA);
+    }
+}
+
+```
+With above implementation, we will have access to both AppStorageV1 and AppStorageV2 which can be used either for migration or used in conjunction as per the use case.
+
+## Rationale
+In order to make the upgrade process as smooth and seamless possible it is mandatory to keep all the components involved in a system as modular as possible. Hence the upgradable storage structure will prove to be very helpful.
+
+
+## Backwards Compatibility
+This standard makes scaling smart contract storage compatible with the future standards & functionality because the storage slots can be added dynamically to enable storage expansion, while also facilitating a means to retrieve and store data across various legacy & in-production storage contracts.
+
+
+## Test Cases
+Test cases are included in the reference implementation refer to Layer3-scaling-standard repository in 0xHashstack's github account
+
+
+## Reference Implementation
+Refer to Layer3-scaling-standard repository in 0xHashstack's github account
+
+
+## Security Considerations
+A storage selector clash occurs when two different storage slot address points to the same location. This has the unintended consequence of overriding data which is unintentional. This scenario is not possible with the diamondCut function prescribed in ([EIP-2535](./eip-2535.md)), because it uses the contract address of facets to generate the slot address.
```

Comment:
```suggestion
A storage selector clash occurs when two different storage slot addresses point to the same location. This has the unintended consequence of overriding data. This scenario is not possible with the `diamondCut` function described in ([EIP-2535](./eip-2535.md)), because it uses the contract address of facets to generate the slot address.
```